rubin_scheduler/utils/consdb.py

```diff
@@ -0,0 +1,1057 @@
+        )
+
+        # Convert to degrees, because everything else is in degrees
+        for body in ("sun", "moon"):
+            for coord in ("alt", "az", "RA", "dec"):
+                column = f"{body}_{coord}"
+                sun_moon[column] = np.degrees(sun_moon[column])
+
+        return sun_moon
+
+    @cached_property
+    def moon_distance(self) -> pd.Series:
+        """Angle between each visit pointing the moon (degrees).
+
+        Returns
+        -------
+        distance : `pd.Series`
+            Angle between each visit pointing the moon (degrees).
+        """
+        # make mypy happy
+        assert isinstance(self.decl.values, np.ndarray)
+        assert isinstance(self.ra.values, np.ndarray)
+
+        moon_ra = self.sun_moon_positions["moon_RA"].values
+        moon_decl = self.sun_moon_positions["moon_dec"].values
+        assert isinstance(moon_ra, np.ndarray)
+        assert isinstance(moon_decl, np.ndarray)
+
+        moon_distance_rad = _angular_separation(
+            np.radians(self.ra.values),
+            np.radians(self.decl.values),
+            np.radians(moon_ra),
+            np.radians(moon_decl),
+        )
+        return pd.Series(np.degrees(moon_distance_rad), index=self.consdb_visits.index)
+
+    @cached_property
+    def solar_elong(self) -> pd.Series:
+        """Solar elongation (degrees) of each visit.
+
+        Returns
+        -------
+        elong : `pd.Series`
+            Solar elongation of each visit in degrees.
+        """
+        # make mypy happy
+        assert isinstance(self.decl.values, np.ndarray)
+        assert isinstance(self.ra.values, np.ndarray)
+
+        sun_ra = self.sun_moon_positions["sun_RA"].values
+        sun_decl = self.sun_moon_positions["sun_dec"].values
+        assert isinstance(sun_ra, np.ndarray)
+        assert isinstance(sun_decl, np.ndarray)
+
+        solar_elong_rad = _angular_separation(
+            np.radians(self.ra.values),
+            np.radians(self.decl.values),
+            np.radians(sun_ra),
+            np.radians(sun_decl),
+        )
+        return pd.Series(np.degrees(solar_elong_rad), index=self.consdb_visits.index)
+
+    @cached_property
+    def zeropoint_e(self) -> pd.Series:
+        """Photometric zero point for each visit, with "counts" in electons.
+
+        Returns
+        -------
+        zp : `pd.Series`
+            Photometric zero point for each visit, with "counts" in electons.
+        """
+        return self.consdb_visits["zero_point_median"] + 2.5 * np.log10(self.gain)
+
+    @cached_property
+    def instrumental_zeropoint_e(self) -> pd.Series:
+        """Instrumental zero point for each visit, with "counts" in electons.
+
+        Returns
+        -------
+        zp : `pd.Series`
+            Instrumental zero point for each visit, with "counts" in electons.
+        """
+        return self.band.apply(self.instrumental_zeropoint_for_band) + 2.5 * np.log10(self.shut_time)
```

Comment:
No, it's not the "predicted zero point", but the zero point due to the instrument, excluding factors outside the instrument (e.g. atmospheric extinction). It's only "predicted" because I have now way to get at "measured."

> Also, is there any reason that both this and the DM zero point have to be per visit

The zero points are per visit because that's how they are reported in the consdb (as near as I could figure out), and I didn't want to introduce an unnecessary inconsistency.

Reply:
(seems like this is used just to calculate the skybrightness in mag/sq arc second?)